app/repository/family.py

```diff
@@ -249,46 +308,69 @@ def search(
     :return list[FamilyReadDTO]: A list of families matching the search
         terms.
     """
-    search = []
-    if "q" in search_params.keys():
+
+    conditions = []
+    params = {}
+
+    # Add conditions based on parameters
+    if "q" in search_params:
         term = f"%{escape_like(search_params['q'])}%"
-        search.append(or_(Family.title.ilike(term), Family.description.ilike(term)))
+        conditions.append("(f.title ILIKE :q OR f.description ILIKE :q)")
+        params["q"] = term
     else:
-        if "title" in search_params.keys():
+        if "title" in search_params:
             term = f"%{escape_like(search_params['title'])}%"
-            search.append(Family.title.ilike(term))
+            conditions.append("f.title ILIKE :title")
+            params["title"] = term
 
-        if "summary" in search_params.keys():
+        if "summary" in search_params:
             term = f"%{escape_like(search_params['summary'])}%"
-            search.append(Family.description.ilike(term))
+            conditions.append("f.description ILIKE :summary")
+            params["summary"] = term
 
     if geography is not None:
-        import_ids = _get_family_import_ids_by_geographies(db, geography)
-        search.append(Family.import_id.in_(import_ids))
+        family_geographies = _get_family_geographies_by_display_values(db, geography)
```

Comment:
suggestion: similar to my other comment - should this just be inline?

If a method doesn't add an abstraction that's reusable or makes it testable - I often question if it should be an abstraction at all.

If anything it just adds a step for our future selves to jump through and having to ask the question of what's the abstraction for if we wanted to refactor.